### ISSUE TITLE
In JS, typedef rename and find-all-refs doesn't work across files

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#23863]
```js
// @Filename: mod1.js
module.exports = { e: 1, m: 1 }
/** @typedef {number} Foo */

// @Filename: use.js
/** @type {import('./mod1').Foo} */
var foo = 12
```

**Expected behavior:**
Renaming `Foo` in use.js should rename it. Find-all-refs on `Foo` in use.js should list all references

**Actual behavior:**
Neither work. However, goto-def does work.

**Comment by andy-ms** [comment]
@sandersn That example doesn't type-check: `module.exports = { e: number, m: number };` fails because `number` is not a value, and `/** @type {Foo} */` fails because `Foo` was never imported. Also, I thought a comment that doesn't precede anything was ignored?

**Comment by sandersn** [comment]
1. Sorry, I typed up the repro based on code in a test buffer. I fixed the type errors.
2. I observed the problem with `import('./mod1').Foo`; I don't know if renaming Foo should work for `const mod1 = require('./mod1'); /** @type {mod1.Foo} */. Probably yes?
3. We have special-case code in the binder for typedef that is supposed to grab every typedef, even at the end of a file. I think they attach to the end of file node or something.
